### PR TITLE
chore: move timeout control to renderer

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

1. Moves the timeout control to the renderer. Now we dont rely in the kernel call, avoiding issue like this #4133 
2. There was a race condition when LOS=1 where the scene was ready before the teleport was requested. This left a scenario were loading was stuck at 0%. This was fixed by checking the scene complete on teleport requested

### <samp>🤖 Generated by Copilot at 4a89752</samp>

Added a timeout feature to the loading screen that notifies the user if the scene is not loaded in time. Refactored and simplified some loading screen and loading status logic using `async` tasks and `CancellationToken`s. Modified `LoadingScreenController.cs` and related files.

## How to test the changes?

1. Test the basic teleport method (goto coordinates, crowd, magic). Loading screen should appear
2. Test the teleport methods from the explorer menu.  Loading screen should appear
3. Change realm without teleporting. Loading screen should not appear
4. Go to different worlds (mgoldman.dcl.eth; dclonboarding.dcl.eth) and back to genesis city. Check that the loading screen appears correctly
5. Select offline in the throtlling devtools console to stop the loading of a scene while the loading screen is on. The timeout should appear after 120 seconds

image

6. Do the signup flow. Everything should work as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs [hunk 1]
```diff
@@ -148,28 +154,39 @@ private bool IsNewRealm()
...
             loadingScreenDataStore.decoupledLoadingHUD.visible.Set(false);
         }
+
+        private void DoTimeout()
+        {
+            notificationsController.ShowNotification(new Model
+            {
+                message = "Loading scene timeout",
+                type = Type.GENERIC,
+                timer = 10f,
```
Comment (eordano) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs:186]: Let's make this 60s
```suggestion
                timer = 60f,
```

**Review by eordano — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs (hunk 1 above)
Comment (dalkia) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs:186]: That one is the time the notification remains on, I think you meant to modify LOAD_SCENE_TIMEOUT, which is currently at 120 seconds. Right?

**Review by popuz — APPROVED** [review]
LGTM. However, stopping with the `devtools->offline` didn't work for me. If I enable offline on "downlading" phase, then 120 timeout do not show up. If I enable it on "loading" phase, then it loads just normally

**Comment by eordano** [comment]
@dalkia merged dev to try out copilot

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs (hunk 1 above)
Comment (eordano) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs:186]: Thanks!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs [hunk 2]
```diff
@@ -113,14 +116,17 @@ private void TeleportRequested(Vector3 current, Vector3 previous)
...
-            }
+                StartTimeoutCounter(timeoutCTS.Token);
+            }else if (IsSceneLoaded(currentDestinationCandidate))
+                HandlePlayerLoading();
+        }
+
+
+        private async UniTaskVoid StartTimeoutCounter(CancellationToken ct)
+        {
+            await UniTask.Delay(LOAD_SCENE_TIMEOUT, cancellationToken: ct);
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs:128]: I would change `LOAD_SCENE_TIMEOUT` to `TimeSpan.FromMilliseconds(LOAD_SCENE_TIMEOUT)` for the sake of readability

**Review by Kinerius — APPROVED** [review]
Nice! This PR is also fixing the Timeout caused by leaving the tab in the background, sometimes it was much less than 120 seconds.

**Review by kwssbocian — APPROVED** [review]
✔️All teleportation methods work correctly.
✔️Changing realms does not cause the player character to be teleported.
✔️Loading screen correctly appears when teleporting to and from worlds.
✔️After 2 minutes of loading the timeout correctly occurs.
Additionally all previous issues regarding incorrect timeout messages have been fixed.

**Pull request merged**